### ISSUE TITLE
Teensy 4.x set system clock implementation

### ISSUE THREAD

**Issue by mjs513** [adafruit/circuitpython#6218]
Started playing again with circuitpython on the Teensy 4.1, and one of things on the todo list was to be able to set the arm clock with the python sketch.  So far I converted the Teensy 4 clocks.c set_arm_clock function to work and compile with circuitpython without errors, updated clocks.h and processor.c in commonbindings.  See: https://forum.pjrc.com/threads/59040-CircuitPython-on-Teensy-4!?p=303483&viewfull=1#post303483 but looks like I have to edit processor.h and processor.c in shared-binding as well.

basically the function in clocks. c is uint32_t setarmclock(uint32_t frequency) but have no clue how to update the code in shared-bindings to accept the function.

FYI:
```
uint32_t common_hal_mcu_processor_setarmclock(uint32_t frequency) {
	SystemCoreClock = setarmclock(frequency);
    return SystemCoreClock;
}
```
Any help would be appreciated - at least point me in the right direction.

**Comment by mjs513** [comment]
Thanks 
figured it out eventually:
```
PU freq: %d 960000000
Pystone(1.2) time for 500 passes = 57769777 ms
This machine benchmarks at 8655 pystones/second


CPU freq: %d 600000000
Pystone(1.2) time for 500 passes = 79864499 ms
This machine benchmarks at 6260 pystones/second
```
